### PR TITLE
Add File Details button

### PR DESCRIPTION
**Pull request by nakedmcse**
Add file details button to admin

### CODE REVIEW

File: src/public/secure/admin.ejs [hunk 1]
```diff
@@ -301,6 +317,28 @@
...
+                                addDetailInfo("File Ext", selectedRow.fileExtension);
+                                addDetailInfo("Original Name", selectedRow.originalFileName);
+                                addDetailInfo("Media Type", selectedRow.mediaType);
+                                addDetailInfo("File Size", selectedRow.fileSize);
+                                addDetailInfo("Created At", customDate()(selectedRow.createdAt, 'display', selectedRow));
+                                addDetailInfo("Expires", selectedRow.expires);
+                                addDetailInfo("IP", selectedRow.ip);
+                                addDetailInfo("URL", selectedRow.url);
+                                const detailsModal = new bootstrap.Modal(document.getElementById('fileDetailsModel'), {keyboard: false});
+                                detailsModal.show();
```
Comment (VictoriqueMoe) [inline src/public/secure/admin.ejs:339]: move this out. i don't think this needs to happen for each time the button is clicked, move it to the `loadDt` function

File: src/public/secure/admin.ejs [hunk 2]
```diff
@@ -470,6 +508,22 @@
         loadDt();
         loadEventListeners();
     });
+
+    function addDetailInfo(heading, detail) {
+        const container = document.getElementById('fileDetailsModelInfoContent');
+        const rowDiv = document.createElement('div');
+        const headingDiv = document.createElement('div');
+        const detailDiv = document.createElement('div');
+        rowDiv.className = 'row';
```
Comment (VictoriqueMoe) [inline src/public/secure/admin.ejs:517]: according to the documentation, a modal requires a `container-fluid` div in that a row is added, considering there will only ever be one `container-fluid`, i recommend added it into `modal-body` and making that new div `fileDetailsModelInfoContent`

File: src/public/secure/admin.ejs [hunk 3]
```diff
@@ -104,6 +104,22 @@
...
+         aria-hidden="true">
+        <div class="modal-dialog" style="max-width:50% !important;">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h1 class="modal-title fs-5" id="fileDetailsModelLabel">File Details</h1>
+                    <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+                </div>
+                <div class="modal-body" id="fileDetailsModelInfoContent">
+                </div>
+                <div class="modal-footer">
```
Comment (VictoriqueMoe) [inline src/public/secure/admin.ejs:118]: you forgot the close button

**Review by VictoriqueMoe — CHANGES_REQUESTED** [review]

File: src/public/secure/admin.ejs [hunk 4]
```diff
@@ -104,6 +104,22 @@
             </div>
         </div>
     </div>
+
+    <div class="modal fade" id="fileDetailsModel" tabindex="-1" aria-labelledby="fileDetailsModelLabel"
+         aria-hidden="true">
+        <div class="modal-dialog" style="max-width:50% !important;">
```
Comment (VictoriqueMoe) [inline src/public/secure/admin.ejs:110]: i don't think we should be adding width, boostrap allows you to size them https://getbootstrap.com/docs/5.3/components/modal/#optional-sizes

**Review by VictoriqueMoe — CHANGES_REQUESTED** [review]

File: src/public/secure/admin.ejs (hunk 3 above)
Comment (nakedmcse) [reply src/public/secure/admin.ejs:118]: Was on purpose - close is X at top - no ?

File: src/public/secure/admin.ejs (hunk 4 above)
Comment (nakedmcse) [reply src/public/secure/admin.ejs:110]: I dislike this, adding the width - will look at the optional sizes

File: src/public/secure/admin.ejs (hunk 1 above)
Comment (nakedmcse) [reply src/public/secure/admin.ejs:342]: Implemented

File: src/public/secure/admin.ejs (hunk 2 above)
Comment (nakedmcse) [reply src/public/secure/admin.ejs:520]: Implemented

File: src/public/secure/admin.ejs (hunk 3 above)
Comment (nakedmcse) [reply src/public/secure/admin.ejs:120]: Changed per request to include close button

File: src/public/secure/admin.ejs (hunk 4 above)
Comment (nakedmcse) [reply src/public/secure/admin.ejs]: Implemented

**Review by VictoriqueMoe — APPROVED** [review]

**Pull request merged**